src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -459,3 +466,61 @@ void function_pointer_restrictionst::write_to_file(
 
   function_pointer_restrictions_json.output(outFile);
 }
+function_pointer_restrictionst function_pointer_restrictionst::merge(
```

Comment:
:bulb: Given that is const, I think a free function that takes two and returns result would be clearer